### PR TITLE
Package/py unicycler

### PR DESCRIPTION
**Pull request by Sinan81**
New package: py-unicycler (bioinformatics)

hybrid assembly pipeline for bacterial genomes

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-unicycler/package.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    homepage = "https://github.com/rrwick/Unicycler"
+    url      = "https://github.com/rrwick/Unicycler/archive/v0.4.5.tar.gz"
+
+    version('0.4.5', 'f7b4f6b712fee6a4fa86a046a6781768')
+
+    depends_on('python@3.4:', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('spades')
+    depends_on('pilon')
+    depends_on('jdk')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-unicycler/package.py]: This is usually a build and/or run dependency. Any idea how it is used by this package?

Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-unicycler/package.py]: Also, I think samtools, racon, .. etc should be run dependencies as well. I should've done some functional testing before committing this I guess! Sorry!

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-unicycler/package.py]: > after attempting running unicycler, I realized setup tools is a link dependency as well

Do you mean run dependency? `setuptools` doesn't have any libraries, just Python files which need to be in your `PYTHONPATH` in order to run.

**Comment by Sinan81** [comment]
my understanding of dependency types is still a bit limited, but I did some functional testing, and the installation now seems to be working.

**Comment by adamjstewart** [comment]
@Sinan81 Have you read:
http://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-types

#3768 provides an even more in-depth look at the meaning of dependency types.

**Comment by alalazo** [comment]
@Sinan81

**Comment by Sinan81** [comment]
Last time I checked the package was compiling and passing examples calculations indicated by the developers. I will get it done very soon.

**Comment by Sinan81** [comment]
Not ready to merge yet ...,

File: var/spack/repos/builtin/packages/py-unicycler/package.py (hunk 1 above)
Comment (Sinan81) [reply var/spack/repos/builtin/packages/py-unicycler/package.py]: made dependency 'jdk' build and run type

**Comment by Sinan81** [comment]
@alalazo I think now the package can be merged. I did some functional testing of the package which confirmed that it was able to detect and use all dependencies. Though I should mention that I load the package via
````
$ source <(spack module tcl loads --dependencies py-unicycler@0.4.7 )
````
When I tried to load the package via the conventional way, dependencies aren't loading:
````
$ spack load python@3.7.2
$ spack activate py-unicycler@0.4.7
````
I am guessing this is due to the fact that python packages are a bit different than regular packages. I am wondering if 'source $< ...' trick is ok. If so, I think it might be a good idea to indicate this in package.py Would you be ok with that?

**Comment by alalazo** [comment]
@Sinan81 From Travis:
```
/home/travis/build/spack/spack/var/spack/repos/builtin/packages/py-unicycler/package.py: has old LGPL license header
==> Error: 1 improperly licensed files
  files with no SPDX-License-Identifier:      0
  files with wrong SPDX-License-Identifier:   0
  files with old license header:              1
```

**Comment by Sinan81** [comment]
@alalazo I am wondering if you can provide input on, in current state of spack, how a python package should be loaded in general.

**Comment by alalazo** [comment]
@Sinan81 You mean in a user environment? What we do at EPFL is to install just a few python packages that need proper optimizations (like `py-numpy`, `py-scipy`, etc.) and activate them, so that the users of our clusters will find them available as soon as they start the interpreter. Of course all of that is from the perspective of somebody maintaining central installations for all the users of a cluster. 

If your use case is more that of a developer who wants to install the dependencies of a software he's working on, I'd have a look at [environments](https://spack.readthedocs.io/en/latest/tutorial_environments.html#environments-tutorial) - I still need to get experienced with those but my impression is that they fit this use case almost perfectly.

**Comment by alalazo** [comment]
> When I tried to load the package via the conventional way, dependencies aren't loading:

You should be able to write in `modules.yaml` that you want your module files to load their dependencies, see [here](https://spack.readthedocs.io/en/latest/tutorial_modules.html#autoload-dependencies).

**Comment by alalazo** [comment]
Thanks!

**Pull request merged**